lib/discourse_code_review/state/github_repo_categories.rb

```diff
@@ -8,8 +8,7 @@ module State::GithubRepoCategories
     class << self
       def ensure_category(repo_name:, repo_id: nil)
         ActiveRecord::Base.transaction(requires_new: true) do
-          repo_category = GithubRepoCategory.find_by(repo_id: repo_id)
-
+          repo_category = GithubRepoCategory.find_by(repo_id: repo_id) if repo_id.present?
```

Comment:
Sorry I might be missing something here but if repo_id was not present, `repo_category` would have been `nil` and then next step we would have found it by name, no?

Maybe your code is more efficient but not sure it's fixing a bug?

Reply:
There are many `GithubRepoCategory` objects which can have multiple `repo_id: nil`. This makes sure it does not select any of these when `repo_id` is not specified.

Reply:
I think (83% sure) they can end up with `nil` when there's a transfer or a rename.

Reply:
> There are many `GithubRepoCategory` objects which can have multiple `repo_id: nil`. This makes sure it does not select any of these when `repo_id` is not specified.

OK I see :+1: